### PR TITLE
Precomposition retracts of maps

### PR DESCRIPTION
**Pull request by fredrik-bakke**
Depends on #973.

- [x] Show that precomp exponentiation of a coherence prism of maps is a coherence prism of maps
- [x] Show that exponentiation of a retract of maps is a retract of maps
- [x] `f`-local types are local at any retract of `f` (this does not require the higher coherence of the prism)

### CODE REVIEW

**Comment by fredrik-bakke** [comment]
I noticed that we have two patterns regarding where we place constructions relating to exponentiating diagrams. For instance `precomp-coherence-square-maps` is placed in `foundation.commuting-squares-of-maps`, while `precomp-coherence-triangle-maps` is placed in `foundation-core.functoriality-function-type`. It seems to me that the operation is more specific to the shape of the diagram than to the functoriality of function types, so I'm moving `precomp-coherence-triangle-maps` to `foundation.commuting-triangles-of-maps`

File: src/orthogonal-factorization-systems/local-types.lagda.md [hunk 1]
```diff
@@ -113,15 +114,69 @@ module _
     map-distributive-Π-is-local-dependent-type (λ a _ → B a)
 ```
 
+### If `S` is `g`-local, and `f` is a retract of `g` then `S` is `f`-local
```
Comment (fredrik-bakke) [inline src/orthogonal-factorization-systems/local-types.lagda.md:117]:
```suggestion
### If `S` is `f`-local then `S` is local at every retract of `f`
```

File: src/foundation/my-file-about-computing-htpy-precomp.lagda.md
Comment (fredrik-bakke) [inline src/foundation/my-file-about-computing-htpy-precomp.lagda.md:1]: So I tried to place the contents of this file in `function-extensionality` or `whiskering-homotopies` or `precomposition-functions`, but couldn't find anywhere to place it without introducing cyclic dependencies. Any advice or ideas is appreciated.

**Comment by fredrik-bakke** [comment]
No pressure, but I would very much appreciate a review of this PR, as I need it for #979 and there are some unresolved issues I could use some advice on.

**Comment by VojtechStep** [comment]
I'll try to take a look at it tomorrow, if I'll have the evening free (that's a 50/50 chance)

**Comment by fredrik-bakke** [comment]
Awesome! Don't sweat it if you don't find the time after all. I have some other things I can work on in the meanwhile.

**Comment by VojtechStep** [comment]
Wait, does this contain #973 ? Then it would make more sense for me to review that first.

**Comment by fredrik-bakke** [comment]
Oh, yeah. you're right. Egbert reviewed that one already though, so I think it is ready for merging.

**Comment by fredrik-bakke** [comment]
If you're not comfortable with merging it without review, I can do it for you ;)

**Comment by fredrik-bakke** [comment]
Lol, it's been a minute since I looked at these PRs

**Comment by VojtechStep** [comment]
I'll give it a quick look and I'll merge it

**Review by VojtechStep — COMMENTED** [review]
I'll wait with reviewing the rest of this PR until we merge #973

**Comment by fredrik-bakke** [comment]
It's not letting me rebase

File: src/foundation/commuting-prisms-of-maps.lagda.md [hunk 2]
```diff
@@ -179,3 +180,203 @@ module _
...
+      ( (i ·l bottom) ·r hA)
+      ( (i ·l g' ·l left) ∙h ((i ·l right) ·r h))) ∙
+    ( ap-binary (_∙_)
+      ( compute-eq-htpy-right-whisk hA (i ·l bottom))
+      ( eq-htpy-concat-htpy (i ·l (g' ·l left)) (i ·l (right ·r h)))) ∙
+    ( ap
+      ( ap (precomp hA S) (eq-htpy (i ·l bottom)) ∙_)
+      ( ap-binary (_∙_)
+        ( ap eq-htpy (eq-htpy (associative-left-whisk-comp i g' left)))
+        ( compute-eq-htpy-right-whisk h (i ·l right))))
```
[the remark this answers is not in the document]
Comment (fredrik-bakke) [reply src/foundation/commuting-prisms-of-maps.lagda.md:257]: I refactored this proof in #989.

**Comment by EgbertRijke** [comment]
> I noticed that we have two patterns regarding where we place constructions relating to exponentiating diagrams. For instance `precomp-coherence-square-maps` is placed in `foundation.commuting-squares-of-maps`, while `precomp-coherence-triangle-maps` is placed in `foundation-core.functoriality-function-type`. It seems to me that the operation is more specific to the shape of the diagram than to the functoriality of function types, so I'm moving `precomp-coherence-triangle-maps` to `foundation.commuting-triangles-of-maps`

This sounds like it would make sense to create files for exponentiating commuting triangles/squares/prisms, and so on the same way we have separate files for other operations

**Comment by fredrik-bakke** [comment]
Sure, that can work. Do you suggest `foundation.precomopsition-commuting-squares-of-maps` and so on?

**Pull request closed without merging**